src/Voron/Data/CompactTrees/CompactKey.cs

```diff
@@ -65,22 +74,18 @@ public void Initialize(LowLevelTransaction tx)
         _currentIdx = 0;
         MaxLength = 0;
 
-        _storage = StoragePool.Rent(2 * Constants.CompactTree.MaximumKeySize);
-        _keyMappingCache = KeyMappingPool.Rent(2 * MappingTableMask);
+        _storage = StoragePool.Get().Rent(2 * Constants.CompactTree.MaximumKeySize);
     }
 
     public void Reset()
     {
-        if (_storage is null || _keyMappingCache is null)
+        if (_storage is null)
             throw new InvalidOperationException("The key has not been initialized before calling reset.");
 
         _owner = null;
 
-        StoragePool.Return(_storage);
+        StoragePool.Get().Return(_storage);
```

Comment:
Can the OS switch the core? If you look into the implementation of `GetCurrentProcessorId`, the comment mentions that it can be refreshed:
https://github.com/dotnet/runtime/blob/21b4a8585362c1bc12d545b63e62a0d9dd4e8673/src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs#L660-L661


So, perhaps it is worth setting `ArrayPool` as a class variable to avoid returning to the wrong pool?

Reply:
There is no ownership semantics in `ArrayPool`. IF that would be the case, then yes, thread hopping would cause those issues.